packages/web/src/routes/session.tsx

```diff
@@ -339,6 +343,16 @@ function Session(props: SessionProps) {
     npmInstall,
   ]);
 
+  useEffect(() => {
+    const handleAutoInstallation = () => {
+      if (autoInstallDependencies) {
+        npmInstall();
```

Comment:
As @BeRecursive22 commented, we have an event we can send up called `deps:install` rather than re-implement the logic here.

This works but for consistency of how we interface with the server, it might be better to dispatch the websocket event. cc @benjreinhart might have an opinion.

Reply:
I agree that it's a better option to install packages once the code cell is created rather than when the component mounts; I will change that.


` npmInstall();`. function executes `deps:install` under the hood,
isn't it better to call  ` npmInstall();` function?  as it does more checks and error handling